src/wallet/components/WalletModal.tsx

```diff
@@ -0,0 +1,261 @@
+import { walletConnectSvg } from '../../internal/svg/walletConnectSvg';
+import {
+  background,
+  border,
+  cn,
+  color,
+  line,
+  pressable,
+  text,
+} from '../../styles/theme';
+import { useOnchainKit } from '../../useOnchainKit';
+
+type WalletModalProps = {
+  isOpen: boolean; // Controls the visibility state of the modal
+  onClose: () => void; // Callback function to close the modal
+  className?: string; // Optional className override for the element
+};
+
+export function WalletModal({ isOpen, onClose, className }: WalletModalProps) {
+  // We need to keep the component mounted during the exit animation
+  const [shouldRender, setShouldRender] = useState(false);
+  const { connect } = useConnect();
+  const connectors = useConnectors();
+  const { config } = useOnchainKit();
+
+  useEffect(() => {
+    if (isOpen) {
+      // Immediately render the modal when opening
+      setShouldRender(true);
+    } else {
+      // When closing, wait for animation to complete before removing from DOM
+      const timer = setTimeout(() => {
+        setShouldRender(false);
+      }, 100);
+      return () => clearTimeout(timer);
+    }
+  }, [isOpen]);
+
+  const appLogo = config?.appearance?.logo;
+  const appName = config?.appearance?.name;
+  const privacyPolicyUrl = config?.wallet?.privacyUrl ?? undefined;
+  const termsOfServiceUrl = config?.wallet?.termsUrl ?? undefined;
+
+  const handleSmartWalletConnector = useCallback(() => {
+    // WalletPreference.SMART_WALLET
+    connect({ connector: connectors[0] });
+    onClose();
+  }, [connect, connectors, onClose]);
+
+  const handleCoinbaseWalletConnector = useCallback(() => {
+    // WalletPreference.EOA
+    connect({ connector: connectors[1] });
+    onClose();
+  }, [connect, connectors, onClose]);
+
+  const handleWalletConnectConnector = useCallback(() => {
+    const walletConnectConnector = connectors.find(
+      (c) => c.name === 'WalletConnect',
+    );
+    if (walletConnectConnector) {
+      connect({ connector: walletConnectConnector });
+      onClose();
+    }
+  }, [connect, connectors, onClose]);
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      onClose();
+    }
+  };
+
+  const handleModalClick = (e: React.MouseEvent) => {
+    e.stopPropagation();
+  };
+
+  // Remove from DOM completely when not rendered
+  if (!shouldRender) {
+    return null;
+  }
+
+  return (
+    <div
+      className={cn(
+        'fixed inset-0 z-50 flex items-center justify-center',
+        isOpen ? 'animate-fadeIn' : 'animate-fadeOut',
+        'bg-black/70',
+        className,
+      )}
+      onClick={onClose}
+      onKeyDown={handleKeyDown}
+      role="presentation"
+      data-testid="modal-overlay"
+    >
+      <div
+        className={cn(
+          border.default,
+          border.radius,
+          background.default,
+          line.default,
+          'w-[323px] p-6 pb-4',
```

Comment:
should this be full screen on mobile?

Reply:
I like how it currently looks on mobile:

<img width="415" alt="Screenshot 2024-11-21 at 9 43 58 AM" src="https://github.com/user-attachments/assets/af93ef72-0fa8-4abd-bf1a-cd9e641301e9">

Reply:
I like the drawer pattern on mobile but obviously that can be a future thing

https://vaul.emilkowal.ski/